### ISSUE TITLE
elpy and aliases

### ISSUE THREAD

**Issue by GPPassos** [jorgenschaefer/elpy#954]
It seems that when I use aliases for packages, elpy-doc and autocompletion won't work correctly.

For instance, using elpy-doc on `numpy.random.normal` works just fine, but using `import numpy as np` and `np.random.normal` doesn't. The behaviour is the same for `from numpy import random as rd` and `rd.normal`.
However, this doesn't happen using `from numpy import random as rd` and elpy-doc on just `rd`.

I'm not sure I've done some mistake. Thank you.

**Comment by gopar** [comment]
Does this only happen to numpy?

I don't have numpy installed but I tried using aliases for packages and it works fine for me. I'm using company-mode instead of auto-complete.

I tried renaming python std libraries and it works. Also tried renaming django and everything still works.

**Comment by jorgenschaefer** [comment]
Elpy heavily relies on other libraries – like Jedi or Rope – for this kind of information. If this does not work in some specific case, it's most likely because the backend you are using can't figure out the aliasing there. I'm afraid there's not much Elpy can do here. You could report this with the backend upstream.

**Comment by vograno** [comment]
I have a similar problem and did some extra experimentation that suggests it might be not a backend problem. Here is a code snippet first:

import numpy as np
np.dtype

1. When the point is right after np.dtype an invocation of elpy-doc does find documentation.
2. When the point is on the blank line at the end of the file an invocation of elpy-doc prompts for a symbol and then it does not find np.dtype. However, it does find numpy.dtype, only the output is formatted differently suggesting it might have come from a different backend of a different call to same backend.

The difference between the two invocations is that in 1. elpy-doc figures the symbol, i.e. np.dtype, by itself, while in 2. the same symbol is passed as an argument.

I observe this difference for both Jedi and Rope backends.

**Comment by jorgenschaefer** [comment]
Yes, when `C-c C-d` prompts for a symbol, it can't figure out the current symbol at point. The information in the Python buffer is not available there.

**Comment by vograno** [comment]
In my example the information WAS available: it was the same buffer, only the point was somewhere else, not next to any symbol. It seems elpy doesn't even TRY to consult the Python buffer when the symbol is supplied manually.
It might be a good idea if elpy always tries to use the Python buffer to resolve the manually supplied symbol. I can't see any downside to this behavior.
If for some reasons this is undesirable maybe we can have a flag that turns this behavior on?

**Comment by jorgenschaefer** [comment]
As far as I know, neither Jedi nor Rope offer the functionality you want. Elpy won't implement this functionality itself, as that's a pretty huge amount of work.
